container/api/src/main/java/org/wildfly/swarm/container/Container.java

```diff
@@ -445,21 +487,59 @@ public Archive createDefaultDeployment() {
                 if (current == null) {
                     factories.put(factory.getType(), factory);
                 } else {
-                    // if this one is high priority than the previously-seen
-                    // factory, replace it.
+                    // if this one is high priority than the previously-seen factory, replace it.
                     if (factory.getPriority() > current.getPriority()) {
                         factories.put(factory.getType(), factory);
                     }
                 }
             }
-            final DefaultDeploymentFactory factory = factories.get(determineDeploymentType());
-
+            
+            DefaultDeploymentFactory factory = factories.get(determineDeploymentType());
             return factory != null ? factory.create() : ShrinkWrap.create(JARArchive.class);
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
     }
 
+    /**
+     * @return The default deployment or null
+     */
+    private Archive<?> getDefaultDeployment() {
+        if (defaultDeployment == null) {
+            try {
+                defaultDeployment = createDefaultDeployment();
+            } catch (RuntimeException ex) {
+                // ignore
+            }
+        }
+        return defaultDeployment;
+    }
+
+    /**
+     * @return The URL to the default deployment or null
+     */
+    private URL getDefaultDeploymentURL() throws IOException {
+        if (defaultDeploymentURL == null) {
+            Archive<?> archive = getDefaultDeployment();
+            if (archive != null) {
+                File tmpdir = TempFileManager.INSTANCE.newTempDirectory("deployment", ".d");
+                new ExplodedExporterImpl(archive).exportExploded(tmpdir);
```

Comment:
This approach would fail in module [arquillian/test](https://github.com/wildfly-swarm/wildfly-swarm-core/tree/master/arquillian/test)